sky/data/storage.py

```diff
@@ -1868,6 +1882,44 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
                     raise exceptions.StorageBucketDeleteError(
                         f'Failed to delete GCS bucket {bucket_name}.')
 
+    def update_labels(self, labels: dict) -> bool:
+        """Assignes given labels to GCS bucket
+        Args:
+          labels: dict; Labels to assign to bucket
+
+        Returns:
+         bool; True if labels are applied sucessfully, False otherwise.
+        """
+        try:
+            bucket_name = self.name
+            bucket = self.client.get_bucket(bucket_name)
+
+            existing_labels = bucket.labels
+
+            if labels is None:
+                labels = {}
+
+            existing_labels.update(labels)
+            bucket.labels = existing_labels
+            bucket.patch()
+            return True
+        except gcp.not_found_exception() as e:
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.StorageBucketGetError(
+                    'Attempted to fetch a non-existent bucket: '
+                    f'{self.name}') from e
+        except (gcp.forbidden_exception(), ValueError) as e:
+            command = f'gsutil ls gs://{self.name}'
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.StorageBucketGetError(
+                    _BUCKET_FAIL_TO_CONNECT_MESSAGE.format(name=self.name) +
+                    f' To debug, consider running `{command}`.') from e
+        except gcp.bad_request_exception() as e:
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.StorageBucketLabelNameError(
+                    f'Can not label bucket: {self.name} with labels: {labels}'
+                ) from e
```

Comment:
Since the labeling for the bucket is a soft requirement, it might be good to `return False` for these exceptions:
```suggestion
        except (gcp.forbidden_exception(), ValueError) as e:
            command = f'gsutil ls gs://{self.name}'
            logger.warning(
                    _BUCKET_FAIL_TO_CONNECT_MESSAGE.format(name=self.name) +
                    f'Error: {common_utils.format_exception(e)}' + 
                    f' To debug, consider running `{command}`.')
        except gcp.bad_request_exception() as e:
            logger.warning(
                f'Can not label bucket: {self.name} with labels: {labels}. '
                f'Error: {common_utils.format_exception(e)'
            )
        return False
```